pkg/cmd/utils/gen-release-notes/render.go

```diff
@@ -0,0 +1,130 @@
+// Copyright (C) 2021 ScyllaDB
+
+package gen_release_notes
+
+import (
+	"fmt"
+	"html/template"
+	"io"
+	"regexp"
+	"strings"
+
+	"github.com/google/go-github/v35/github"
+)
+
+const releaseNotesTemplate = `
+# Release notes for {{ .Release }}
+
+## Changes By Kind
+
+{{- range $prKind := .PRKinds }}
+  {{- with $filteredPRs := ( FilterByLabelKinds $.PRs $prKind.Kinds ) }}
+
+### {{ $prKind.Title }}
+
+    {{- range $pr := $filteredPRs }}
+{{ FormatPR $pr }}
+    {{- end }}
+  {{- end }}
+{{- end }}
+
+{{- with $uncategorized := ( Uncategorized .PRs ) }}
+### Uncategorized
+
+  {{- range $pr := $uncategorized }}
+{{ FormatPR $pr }}
+  {{- end }}
+{{- end }} 
+
+---
+`
+
+var branchPrefixRe = regexp.MustCompile(`^(\[[^]].*])?(.*)`)
+
+func cleanupPRTitle(title string) (string, error) {
+	m := branchPrefixRe.FindStringSubmatch(strings.TrimSpace(title))
+	if len(m) != 3 {
+		return "", fmt.Errorf("%q PR title cannot be parsed", title)
+	}
+	return m[2], nil
+}
+
+func formatPR(pr *github.PullRequest) (string, error) {
+	title, err := cleanupPRTitle(pr.GetTitle())
+	if err != nil {
+		return "", err
+	}
+	user := pr.GetUser()
+	details := fmt.Sprintf("[#%d](%s),[@%s](%s)", pr.GetNumber(), pr.GetHTMLURL(), user.GetLogin(), user.GetHTMLURL())
+	return fmt.Sprintf("* %s (%s)", title, details), nil
+}
+
+type prKind struct {
+	Title string
+	Kinds []string
+}
+
+type releaseNoteData struct {
+	PRKinds []prKind
+	PRs     []*github.PullRequest
+	Release string
+}
+
+func renderReleaseNotes(out io.Writer, release string, pullRequests []*github.PullRequest) error {
+	data := releaseNoteData{
+		Release: release,
+		PRs:     pullRequests,
+		PRKinds: []prKind{
+			{Title: "API Changes", Kinds: []string{"api-change"}},
+			{Title: "Features", Kinds: []string{"feature"}},
+			{Title: "Bugs", Kinds: []string{"bug"}},
+			{Title: "Documentation", Kinds: []string{"documentation"}},
+			{Title: "Flaky/Failing Tests", Kinds: []string{"failing-test", "flake"}},
+			{Title: "Other", Kinds: []string{"cleanup"}},
+		},
+	}
+
+	t := template.New("release-notes").Funcs(map[string]interface{}{
+		"FilterByLabelKinds": filterByLabelKinds,
```

Comment:
I guess you could just create a map kinds->prs and save some logic in the template but I don't feel strong